api/api.go

```diff
@@ -176,7 +176,7 @@ func entry(resource string, options *EntryOptions) (*base.SentinelEntry, *base.B
 		// must finish the lifecycle of r.
 		blockErr := base.NewBlockErrorFromDeepCopy(r.BlockError())
 		e.Exit()
-		return nil, blockErr
+		return e, blockErr
```

Comment:
Why return entry when blocked, and what do you need to do with it

Reply:
Leave the control to caller, and they can do any thing like log/metrics/fallback

Reply:
I don't understand. Do you want to get an EntryContext through entry? or you can give an example, how do you do things like log/metric/fallback

Reply:
Here is the use case. Right now, there is no way to fetch entry info like ResourceType/TrafficType... when error happens.

https://github.com/douyu/jupiter/blob/493bb847aeb6fd28addbd6c7d621c34f0f2a2819/pkg/core/sentinel/config.go#LL103